tidb-cloud/setup-self-hosted-kafka-pls.md

```diff
@@ -0,0 +1,759 @@
+
+    - Broker target group 3
+
+        - **Target type**: `Instances`
+        - **Target group name**: `broker-target-group-3`
+        - **Protocol**: `TCP`
+        - **Port**: `9095`
+        - **IP address type**: `IPv4`
+        - **VPC**: `Kafka VPC`
+        - **Health check protocol**: `TCP`
+        - **Register targets**: `broker-node3:39092`
+
+2. Go to [Load balancers](https://console.aws.amazon.com/ec2/home#LoadBalancers:) to create a network load balancer.
+
+    - **Load balancer name**: `kafka-lb`
+    - **Schema**: `Internal`
+    - **Load balancer IP address type**: `IPv4`
+    - **VPC**: `Kafka VPC`
+    - **Availability Zones**: 
+        - `usw2-az1` with `broker-usw2-az1 subnet`
+        - `usw2-az2` with `broker-usw2-az2 subnet`
+        - `usw2-az3` with `broker-usw2-az3 subnet`
+    - **Security groups**: create a new security group with the following rules.
+        - Inbound rule allows all TCP from Kafka VPC: Type - `All TCP`; Source - `Anywhere-IPv4`
+        - Outbound rule allows all TCP to Kafka VPC: Type - `All TCP`; Destination - `Anywhere-IPv4`
+    - Listeners and routing:
+        - Protocol: `TCP`; Port: `9092`; Forward to: `bootstrap-target-group`
+        - Protocol: `TCP`; Port: `9093`; Forward to: `broker-target-group-1`
+        - Protocol: `TCP`; Port: `9094`; Forward to: `broker-target-group-2`
+        - Protocol: `TCP`; Port: `9095`; Forward to: `broker-target-group-3`
+
+3. Test the load balancer in the bastion node. This example only tests the Kafka bootstrap. Because the load balancer is listening on the Kafka EXTERNAL listener, the addresses of EXTERNAL advertised listeners can not be resolved in the bastion node. Note down the `kafka-lb` DNS name from the load balancer detail page, for example `kafka-lb-77405fa57191adcb.elb.us-west-2.amazonaws.com`. Execute the script in the bastion node.
+
+    ```shell
+    # Please replace {lb_dns_name} to the actual value
+    export JAVA_HOME=/home/ec2-user/jdk-22.0.2
+    ./kafka_2.13-3.7.1/bin/kafka-broker-api-versions.sh --bootstrap-server {lb_dns_name}:9092
+
+    # Expected output(last 3 lines), order may be different.
+    b1.usw2-az1.abc.us-west-2.aws.3199015.tidbcloud.com:9093 (id: 1 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
+    b2.usw2-az2.abc.us-west-2.aws.3199015.tidbcloud.com:9094 (id: 2 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
+    b3.usw2-az3.abc.us-west-2.aws.3199015.tidbcloud.com:9095 (id: 3 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
+
+    # You can also try bootstrap in others ports 9093/9094/9095. It will succeed probabilistically since NLB in AWS resolve LB DNS to IP address of any availability zone and disable cross-zone load balancing by default. 
+    # If you enable cross-zone load balancing in LB, it will be certainly success. But it is unnecessary, and might cause potential cross-AZ traffic.
```

Comment:
```suggestion
    # You can also try bootstrap in other ports 9093/9094/9095. It will succeed probabilistically because NLB in AWS resolves LB DNS to the IP address of any availability zone and disables cross-zone load balancing by default. 
    # If you enable cross-zone load balancing in LB, it will succeed. However, it is unnecessary and might cause additional cross-AZ traffic.
```